src/test/java/ClassUtils.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+import org.mule.runtime.module.artifact.api.classloader.ClassLoaderRepository;
+
+import java.lang.reflect.Field;
+
+/**
+ * Provides utilities related to {@link Class} instances needed for tests due to DataWeave limitations.
+ */
+public class ClassUtils {
+
+  private ClassUtils() {}
+
+  public static Field getDeclaredField(ClassLoader classLoader, String className, String fieldName) {
+    try {
+      Class<?> type = classLoader.loadClass(className);
+      return type.getDeclaredField(fieldName);
+    } catch (ClassNotFoundException | NoSuchFieldException e) {
+      throw new RuntimeException(e);
```

Comment:
We're basically using this only in MUnits, in order to use the connector class loader to load the shaded classes with the modified fields.

Aside from the possible misuse in an MUnit test, if the shading plugin is removed from the connector configuration, or changed to a different package, invoking this method during test assertions will result in an exception being thrown, and therefore a test failure.

While the overall goal is to ensure that the library shading isn't broken in the future, here is what a misconfiguration might look like:

```xml
<global-property name="shadePackage" value="org.mule.module.json.internal.shaded.INVALID"/>
<munit:parameterization name="JsonNodeReader">
    <munit:parameters>
        <munit:parameter propertyName="className" value="com.github.fge.jackson.INVALIDJsonNodeReader"/>
        <munit:parameter propertyName="memberName" value="INVALID_NAME"/>
        <munit:parameter propertyName="expectedModifiers" value="public static"/>
    </munit:parameters>
</munit:parameterization>
```

Since we're just using this code as a means to access the mule internal class loader without dealing with internal package issues, we can just add these exceptions to the method signature instead of handing anything, because the `ClassNotFoundException`, `NoSuchFieldException` exception names are fairly descriptive.